starknet_py/net/gateway_client.py

```diff
@@ -53,17 +54,23 @@ class GatewayClient(Client):
     def __init__(
         self,
         net: Network,
-        chain: StarknetChainId = None,
+        chain: Optional[StarknetChainId] = None,
         session: Optional[aiohttp.ClientSession] = None,
     ):
         """
         Client for interacting with starknet gateway.
 
         :param net: Target network for the client. Can be a string with URL or one of ``"mainnet"``, ``"testnet"``
-        :param chain: Chain used by the network. Required if you use a custom URL for ``net`` param.
+        :param chain: Chain used by the network. Required if you use a custom URL for ``net`` param. Chain is deprecated
+                        and will be removed in the next releases
         :param session: Aiohttp session to be used for request. If not provided, client will create a session for
                         every request. When using a custom session, user is resposible for closing it manually.
         """
+        if chain is not None:
+            warnings.warn(
+                "Chain is deprecated and will be deleted in the next releases"
+            )
```

Comment:
I'm not sure if we should be adding these deprecation warnings to the constructors. If the user isn't using them anyway, they'll just have to remove this parameter once we deprecate chain already.

It's removal is only problem for the users using chain property.